p2p/3.IPFS_or_HTTP/README.md

```diff
@@ -0,0 +1,149 @@
+web_1  | [06/Aug/2021 11:02:59] "GET /media/static/Scott_Holmes_Music_-_Country_Road_Drive.mp3 HTTP/1.1" 200 8983053
+```
+Let me explain:
+the first part of the message is obviously the datetime. The second one is the method, `GET`, because we want to just
+get the page ; we are asking the server to give us the `/` route which is the home page.
+Then, the `HTTP` protocol and finally two numbers. The `200` is the most interesting : it is a status, preview code. `200` means
+that the server is ok to give us that page from the `/` route, and it has been delivered correctly.
+
+>What does `POST` means ?
+
+`POST` is another **method** than `GET`. When you fill the form earlier, it was **you** that was giving the server
+some information :that is the mainly difference between `POST` and `GET`.
+You can find all HTTP methods [here](https://www.restapitutorial.com/lessons/httpmethods.html).
+
+Now, look at your `/media/static` folder : you have the mp3 file you just downloaded in here !
+>This is how HTTP works. When retrieving data, HTTP focuses on **location**. Have you noticed the url ?
+First, it is 0.0.0.0 which is your local IP; then :8000 to signify the port. Finally, the routes or file you want to
+get, joined by a /. That format does not ring a bell to you ? It is like a path !
+
+## Step 2 - IPFS
+
+>HTTP is cool but has its limits : if the server is down, you won't be able to retrieve the data stored. Furthermore,
+your government can easily block access to certain servers by their IPs that host particular website for censure purposes.
+Let's see how IPFS answers this issues.
+
+At its core, IPFS is a **distributed** system for storing and accessing files, websites, applications, and data.
+Instead of referring to data (photos, articles, videos) by **location**, or which server they are stored on, IPFS refers to everything by that data’s hash, meaning the **content itself.**
+
+The idea is that if you want to access a particular page from your browser, IPFS will ask the entire network, “does anyone have the data that corresponds to this hash?” A node on IPFS that contains the corresponding hash will return the data, allowing you to access it from anywhere (and potentially even offline).
+
+If this is not enough clear for you, I strongly advise you to refer to [this](https://www.youtube.com/watch?v=5Uj6uR3fp-U) video.
+
+### 2.0 Improve the storage
+
+Here is what we are going to do : We are going to upload our files directly on IPFS and not locally anymore.
+Instead of having the file locally, let's have its corresponding hash in our database.
+
+1. Go to `musicshare/models.py` and add a CharField for the hash.
+2. Go to `musicshare/views.py` and modify the code of the upload view to communicate with IPFS API in order to upload the file there.
+   Please use port `5001` for the connection.
+3. Open `musicshare/templates/musicshare/index.html` to line `179` and make sure the hash of the song appears.
```

Comment:
Replace path by a link
Example
`./toto` -> [./toto](./toto)

This way, they can directly click on the link to go to the file